python/ml4ir/base/pipeline.py

```diff
@@ -351,17 +301,62 @@ def run(self):
                     pad_sequence=self.args.pad_sequence_at_inference,
                 )
 
-            # Finish
-            self.finish()
-
         except Exception as e:
             self.logger.error("!!! Error Training Model: !!!\n{}".format(str(e)))
             traceback.print_exc()
-            job_status = ("_FAILURE", "{}\n{}".format(str(e), traceback.format_exc()))
+            job_status = "_FAILURE"
+            job_info = "{}\n{}".format(str(e), traceback.format_exc())
+
+        # Write experiment tracking data in job status file
+        experiment_tracking_dict = dict()
+
+        # Add command line script arguments
+        experiment_tracking_dict.update(vars(self.args))
 
+        # Add feature config information
+        experiment_tracking_dict.update(self.feature_config.get_hyperparameter_dict())
+
+        # Add train and test metrics
+        experiment_tracking_dict.update(train_metrics)
+        experiment_tracking_dict.update(test_metrics)
+
+        job_info = pd.DataFrame.from_dict(
+            experiment_tracking_dict, orient="index", columns=["value"]
+        ).to_csv()
+
+        # Finish
+        self.finish(job_status, job_info)
+
+    def finish(self, job_status, job_info):
+        """
```

Comment:
We need to add pydoc about what finish does.

Reply:
Added.